File: ai/src/main/java/org/pingel/causality/examples/Model3dot9h.java

package org.pingel.causality.examples;

import org.pingel.bayes.CausalModel;
import org.pingel.bayes.Function;
import org.pingel.bayes.ModelVisualizer;
import org.pingel.bayes.RandomVariable;

public class Model3dot9h extends CausalModel {

    public Model3dot9h()
    {
        super("3.9h");

        RandomVariable W = addVariable(new RandomVariable("W", "w"));
        RandomVariable X = addVariable(new RandomVariable("X", "x"));
		RandomVariable Y = addVariable(new RandomVariable("Y", "y"));
		RandomVariable Z = addVariable(new RandomVariable("Z", "z"));
		RandomVariable U1 = addVariable(new RandomVariable("U1", "u1", false));
		RandomVariable U2 = addVariable(new RandomVariable("U2", "u2", false));
		RandomVariable U3 = addVariable(new RandomVariable("U3", "u3", false));
		RandomVariable U4 = addVariable(new RandomVariable("U4", "u4", false));

		addFunction(new Function(W, X, U3));
		addFunction(new Function(X, Z, U1, U2));
		addFunction(new Function(Y, W, U2, U4));
		addFunction(new Function(Z, U1, U3, U4));

    }

    public static void main(String[] argv)
    {
        CausalModel model = new Model3dot9h();
        ModelVisualizer.draw(model);
    }

}

File: ai/src/main/scala/org/pingel/causality/examples/Model3dot9h.scala

package org.pingel.causality.examples

import org.pingel.causality.CausalModel
import org.pingel.forms.Basic.PFunction
import org.pingel.bayes.RandomVariable

object Model3dot9h extends CausalModel("3.9h") {

  val W = new RandomVariable("W")
  val X = new RandomVariable("X")
  val Y = new RandomVariable("Y")
  val Z = new RandomVariable("Z")
  val U1 = new RandomVariable("U1", None, false)
  val U2 = new RandomVariable("U2", None, false)
  val U3 = new RandomVariable("U3", None, false)
  val U4 = new RandomVariable("U4", None, false)

  g ++= (W :: X :: Y :: Z :: U1 :: U2 :: U3 :: U4 :: Nil)
  
  addFunction(new PFunction(W, List(X, U3)))
  addFunction(new PFunction(X, List(Z, U1, U2)))
  addFunction(new PFunction(Y, List(W, U2, U4)))
  addFunction(new PFunction(Z, List(U1, U3, U4)))

  def main(args: Array[String]) {
    g.draw
  }

}
